sklearn/cross_validation.py

```diff
@@ -60,6 +61,11 @@
            'permutation_test_score',
            'train_test_split']
 
+@deprecated("FitFailedWarning has been moved into the sklearn.exceptions"
+            " module. It will not be available here from version 0.19")
+class FitFailedWarning(_FitFailedWarning):
```

Comment:
What is the point of providing a way to import FitFailedWarning from the deprecated cross_validation module? It feels like something quite internal. Are they use cases where people would like to import this warning?

Also, your current implementation doesn't allow you to use sklearn.cross_validation.FitFailedWarning the same way as sklearn.exceptions.FitFailedWarning. A snippet showing what I mean:

```python
import numpy as np

from sklearn.utils.testing import assert_warns

from sklearn.datasets import make_classification
from sklearn.exceptions import FitFailedWarning

from sklearn.base import BaseEstimator
from sklearn.grid_search import GridSearchCV
from sklearn.cross_validation import (FitFailedWarning
                                      as DeprecatedFitFailedWarning)


class FailingClassifier(BaseEstimator):
    """Classifier that raises a ValueError on fit()"""

    FAILING_PARAMETER = 2

    def __init__(self, parameter=None):
        self.parameter = parameter

    def fit(self, X, y=None):
        if self.parameter == FailingClassifier.FAILING_PARAMETER:
            raise ValueError("Failing classifier failed as required")

    def predict(self, X):
        return np.zeros(X.shape[0])

# GridSearchCV with on_error != 'raise'
# Ensures that a warning is raised and score reset where appropriate.

X, y = make_classification(n_samples=20, n_features=10, random_state=0)

clf = FailingClassifier()

# refit=False because we only want to check that errors caused by fits
# to individual folds will be caught and warnings raised instead. If
# refit was done, then an exception would be raised on refit and not
# caught by grid_search (expected behavior), and this would cause an
# error in this test.
gs = GridSearchCV(clf, [{'parameter': [0, 1, 2]}], scoring='accuracy',
                  refit=False, error_score=0.0)

assert_warns(FitFailedWarning, gs.fit, X, y)  # this assert passes

assert_warns(DeprecatedFitFailedWarning, gs.fit, X, y)  # this assert fails
```

This problem can be fixed by using something like this:

```python
MyDeprecatedFitFailedWarning = deprecated('Bla')(FitFailedWarning)
```